crates/common/src/raindex_client/local_db/fetch.rs

```diff
@@ -119,14 +119,108 @@ impl LocalDb {
         all_events.sort_by(|a, b| {
             let block_a = extract_block_number_from_entry(a).unwrap_or(0);
             let block_b = extract_block_number_from_entry(b).unwrap_or(0);
-            block_a.cmp(&block_b)
+            block_a
+                .cmp(&block_b)
+                .then_with(|| a.log_index.cmp(&b.log_index))
         });
 
         self.backfill_missing_timestamps(&mut all_events, config)
             .await?;
         Ok(all_events)
     }
 
+    pub async fn fetch_store_set_events(
+        &self,
+        store_addresses: &[String],
+        start_block: u64,
+        end_block: u64,
+        config: &FetchConfig,
+    ) -> Result<Vec<LogEntryResponse>, LocalDbError> {
+        if store_addresses.is_empty() {
+            return Ok(Vec::new());
+        }
+
+        let unique_addresses: Vec<String> = {
+            let mut dedup = HashSet::new();
+            store_addresses
+                .iter()
+                .filter_map(|addr| {
+                    let lower = addr.to_ascii_lowercase();
+                    if dedup.insert(lower.clone()) {
+                        Some(lower)
+                    } else {
+                        None
+                    }
+                })
+                .collect()
+        };
```

Comment:
consider itertool's `.unique` (NOTE: the following will make all addresses lowercase. this becomes even simpler when using `Address` over `String`)

```suggestion
        let unique_addresses: Vec<String> = store_addresses
            .iter()
            .map(|addr| addr.to_ascii_lowercase())
            .unique()
            .collect();
```